### PR TITLE
Add mkcloud step to presetup the crowbar development setup

### PR DESCRIPTION
**Pull request by rsalevsky**
This changes makes it easy for all crowbar developers to use the crowbar
development setup. Specially for newcommers this change is important.

### CODE REVIEW

**Comment by nicolasbock** [comment]
Nice idea! +1

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -639,17 +639,28 @@ function addcloud6pool()
...
 function addcctdepsrepo()
 {
+    if [[ "$cloudsource" == @(develcloud5|GM5|GM5+up) ]]; then
+        zypper ar -f http://$susedownload/ibs/Devel:/Cloud:/Shared:/Rubygem/SLE_11_SP3/Devel:Cloud:Shared:Rubygem.repo
+    else
+        add_sdk_repo
+    fi
+}
+
+function add_sdk_repo()
```
Comment (aspiers) [inline scripts/qa_crowbarsetup.sh]: I like this "extract function" refactoring. Personally I would have put it in a separate commit for the reasons stated in https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes.

**Comment by aspiers** [comment]
Yes, nice idea.  Where is this new step documented?

**Comment by aspiers** [comment]
Actually, shouldn't it be documented in `docs/mkcloud.md`?  If so, you can easily turn my gentle -1 into a strong +1 by doing that ;-)

**Comment by rsalevsky** [comment]
@aspiers I will adapt [development.md](https://github.com/crowbar/crowbar/blob/master/doc/development.md) once this is merged. We will also have a discussion about the mkcloud documentation while the workshop.

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -4443,6 +4454,42 @@ function onadmin_run_cct()
...
+    add_sdk_repo
+    ensure_packages_installed git gcc ruby2.1-devel sqlite3-devel libxml2-devel
+
+    # create development folders
+    mkdir -p /opt/crowbar/crowbar_framework/db /opt/crowbar/barclamps
+
+    # copy existing database
+    cp /opt/dell/crowbar_framework/db/production.sqlite3 /opt/crowbar/crowbar_framework/db/development.sqlite3
+
+    # clone git repos
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: I prefer to work on my local machine instead of the admin node directly, as I don't want all my local git branches/stashes etc to just disappear when I reset the cloud. that was one of the benefits why @tboerger has written this guard script

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -4443,6 +4454,42 @@ function onadmin_run_cct()
...
+    local crowbar_git_dir=/opt/crowbar/git/crowbar
+    git clone https://github.com/crowbar/crowbar.git $crowbar_git_dir
+    git clone https://github.com/crowbar/crowbar-core.git $crowbar_git_dir/barclamps/core
+    git clone https://github.com/crowbar/crowbar-openstack.git $crowbar_git_dir/barclamps/openstack
+    git clone https://github.com/crowbar/crowbar-ceph.git $crowbar_git_dir/barclamps/ceph
+    git clone https://github.com/crowbar/crowbar-ha.git $crowbar_git_dir/crowbar/barclamps/ha
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: why using another /crowbar/ subdirectory here?

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -4443,6 +4454,42 @@ function onadmin_run_cct()
...
+
+    # install development gems and generate dir tree
+    pushd $crowbar_git_dir
+    bundle install
+    GUARD_SYNC_HOST=localhost bundle exec guard
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: again, using guard from my host is really preferable. and this is just a one shot, and then you call it again in the foreground when the cloud is installed? it's not really visible or documented from here like in https://github.com/SUSE-Cloud/automation/pull/712/files#diff-4da4ef75e4826a188f831da36089295cR1095

**Comment by MaximilianMeister** [comment]
overall I still don't get why you have duplicated the work of https://github.com/SUSE-Cloud/automation/pull/712 and not using locally maintained git clones, instead cloning them on every setup and throwing them away in the end I find not really efficient

**Comment by rsalevsky** [comment]
We discussed the idea of this PR offline, for the record:

We use this PR as base for #712 to have something which runs without local data to avoid the issues with the mkcloud hosts.

File: scripts/qa_crowbarsetup.sh (hunk 4 above)
Comment (rsalevsky) [inline scripts/qa_crowbarsetup.sh]: The guard here is just used to create the initial directory tree. After the setup is complete I prefer to use the local guard to sync the stuff.

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (rsalevsky) [inline scripts/qa_crowbarsetup.sh]: We need this git clones to use guard to setup the initial folder structure which is needed to install all GEM's and the barclamps.

**Comment by aspiers** [comment]
> We use this PR as base for #712 to have something which runs without local data to avoid the issues with the mkcloud hosts.

What do you mean by "local data" and "issues with the mkcloud hosts"?  Thanks for helping me understand :)

**Comment by rsalevsky** [comment]
With local data I mean the crowbar checkouts from GitHub. Those are needed to create the dev environment see the `host_crowbar_dev_dir` var in #712. That means without the checkouts you are not able to use mkcloud to setup a dev environment with #712. Specially on our mkcloud hosts this can be painful see https://github.com/SUSE-Cloud/automation/pull/712#issuecomment-165037965.

To avoid all this issues this approach don't require any local data.

File: scripts/qa_crowbarsetup.sh (hunk 4 above)
Comment (tboerger) [inline scripts/qa_crowbarsetup.sh]: But this guard will not exit.

Comment (rsalevsky) [inline scripts/qa_crowbarsetup.sh]: Exactly, so it will create the initial sync with the correct folder tree (bin, barclamps, crowbar_framwork, ...) and the script will continue.

Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: > But this guard will not exit.

@tboerger `qa_crowbarsetup.sh` will continue somehow. I had the same issue here https://github.com/SUSE-Cloud/automation/pull/712/files#diff-4da4ef75e4826a188f831da36089295cR670 where I had to redirect to keep it in the foreground.

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -639,17 +639,28 @@ function addcloud6pool()
...
+    else
+        add_sdk_repo
+    fi
+}
+
+function add_sdk_repo()
+{
     case "$cloudsource" in
-        develcloud5|GM5|GM5+up)
-            zypper ar -f http://$susedownload/ibs/Devel:/Cloud:/Shared:/Rubygem/SLE_11_SP3/Devel:Cloud:Shared:Rubygem.repo
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: why is this dropped, when we only just EOLed Cloud4? or is this never used?

Comment (rsalevsky) [inline scripts/qa_crowbarsetup.sh]: Yes, I don't want to mix this.

**Comment by MaximilianMeister** [comment]
+1
